lib/pbench/server/api/resources/query_apis/query_dataset.py

```diff
@@ -0,0 +1,268 @@
+        to search.
+
+        "end" is a time string representing the end range of dates to
+        to search.
+
+        The Elasticsearch URL string is also dependent upon the configured
+        "prefix". This is a Pbench artifact that allows multiple Pbench
+        instances to share a single Elasticsearch server, by segregating
+        their data into unique namespaces. Each Pbench index name is
+        qualified by the index_prefix of the Pbench server that originally
+        indexed the data. The prefix comes from the pbench-server.cfg
+        file for the server instance. For example, production, "staging",
+        and development server instances might all share one Elasticsearch
+        cluster by choosing unique prefixes to generate unique index
+        names.
+
+        Required headers include
+
+        X-auth-token:   Pbench session authorization token authorized to access
+                        "user"'s data. E.g., "user", "admin" or a user with
+                        whom the dataset has been shared.
+        Content-Type:   application/json
+        Accept:         application/json
+
+        Return payload is a summary of the "aggregations" property of the
+        returned Elasticsearch query results, showing fields from the "run"
+        subdocument, including the controller and name, the start and end
+        time, and a starting UNIX timestamp.
+
+        TODO: We probably need to return all *unowned* runs when called
+        without a session token or user parameter. We need to define precisely
+        how this should work.
+
+        NOTE: This is the format currently constructed by the Pbench
+        dashboard `src/model/dashboard.js` fetchResults method, which
+        becomes part of the Redux state. The "runMetadata" is a join of
+        the ES run document's "@metadata" and "run" sub-documents, while
+        "hostTools" is the run document's "host_tools_info" nested array.
+
+        [
+            {
+                "runMetadata": {
+                    "file-name": "/pbench/archive/fs-version-001/dhcp31-187.perf.lab.eng.bos.redhat.com/fio_rhel8_kvm_perf43_preallocfull_nvme_run4_iothread_isolcpus_2020.04.29T12.49.13.tar.xz",
+                    "file-size": 216319392,
+                    "md5": "12fb1e952fd826727810868c9327254f",
+                    [...]
+                },
+                "hostTools": [
+                    {
+                        "hostname": "dhcp31-187",
+                        "tools": {
+                            "iostat": "--interval=3",
+                            [...]
+                        }
+                    }
+                ]
+            }
+        ]
+        """
+        json_data = request.get_json(silent=True)
+        if not json_data:
+            self.logger.info("Invalid JSON object. Query: {}", request.url)
+            abort(400, message="Missing request payload")
+
+        try:
+            user = json_data["user"]
```

Comment:
If we have an authorization header included we could deduce the username and email field from it and a request json payload does not necessarily need user information.

Reply:
Yes, we can deduce from the token, but I didn't have that when I wrote this, and we still haven't wired in your authorization decorators. This was partly a way to move forward without all that; but we also need to deal with whether an authorized user sees just their own data or/also public data, and `"user"` provides a hook to override the default. I also wanted to open the possibility for an admin user to see any user's data.